src/asmcnc/skavaUI/popup_info.py

```diff
@@ -514,310 +516,376 @@ def __init__(self, screen_manager, localization, description):
-                    "\n" + \
-                    self.l.get_str("Please do not restart your machine until you are prompted to do so.")
+                      "\n\n" + \
+                      self.l.get_str("Update message") + ": " + \
+                      message + \
+                      "\n" + \
+                      self.l.get_str("Please do not restart your machine until you are prompted to do so.")
 
         title_string = self.l.get_str('Update Successful!')
         ok_string = self.l.get_bold('Ok')
 
         def reboot(*args):
             self.sm.current = 'rebooting'
-                    
+
         img = Image(source="./asmcnc/apps/shapeCutter_app/img/info_icon.png", allow_stretch=False)
-        label = Label(size_hint_y=1.2, text_size=(660, None), halign='center', valign='middle', text=description, color=[0,0,0,1], padding=[40,10], markup = True)
-   
-        ok_button = Button(text=ok_string, markup = True)
+        label = Label(size_hint_y=1.2, text_size=(660, None), halign='center', valign='middle', text=description,
+                      color=[0, 0, 0, 1], padding=[40, 10], markup=True)
+
+        ok_button = Button(text=ok_string, markup=True)
         ok_button.background_normal = ''
         ok_button.background_color = [76 / 255., 175 / 255., 80 / 255., 1.]
-      
-        btn_layout = BoxLayout(orientation='horizontal', spacing=10, padding=[0,0,0,0])
+
+        btn_layout = BoxLayout(orientation='horizontal', spacing=10, padding=[0, 0, 0, 0])
         btn_layout.add_widget(ok_button)
-        
-        layout_plan = BoxLayout(orientation='vertical', spacing=10, padding=[40,20,40,20])
+
+        layout_plan = BoxLayout(orientation='vertical', spacing=10, padding=[40, 20, 40, 20])
         layout_plan.add_widget(img)
         layout_plan.add_widget(label)
         layout_plan.add_widget(btn_layout)
-        
+
         popup = Popup(title=title_string,
                       title_color=[0, 0, 0, 1],
-                      title_font= 'Roboto-Bold',
-                      title_size = '20sp',
+                      title_font='Roboto-Bold',
+                      title_size='20sp',
                       content=layout_plan,
                       size_hint=(None, None),
                       size=(700, 400),
-                      auto_dismiss= False
+                      auto_dismiss=False
                       )
-        
+
         popup.separator_color = [249 / 255., 206 / 255., 29 / 255., 1.]
         popup.separator_height = '4dp'
         popup.background = './asmcnc/apps/shapeCutter_app/img/popup_background.png'
-        
+
         ok_button.bind(on_press=popup.dismiss)
         # ok_button.bind(on_press=reboot)
-        
+
         popup.open()
-        
+
         Clock.schedule_once(reboot, 6)
-    
-class PopupSoftwareRepair(Widget):   
+
+# Popup asking the user if they are sure they want to continue with the update with a given warning_message
+class PopupSoftwareUpdateWarning(Widget):

```

Comment:
Because this is a brand new popup, and one that will need full translation checking, it would make sense to write a proper test script for visually checking it! This also makes development much easier, especially when designing more complex UIs, so it's good to get in the habit early :) 

[Example here](https://github.com/YetiTool/easycut-smartbench/blob/master/tests/manual_tests/visual_screen_tests/yetipilot_settings_popup_test.py)